File: May-LeetCoding-Challenge/02-Jewels-And-Stones/Jewels-And-Stones.java
class Solution {
    public int numJewelsInStones(String J, String S) {
        if (J == null || S==null) return 0;
        Set<Character> jmap = new HashSet<>();
        
        //time O(len of J)
        //space O(len of J)
        for (int i = 0; i<J.length(); i++){
            jmap.add(J.charAt(i));
        }
        
        //time O(len of S)
        int count = 0;
        for(int i = 0; i<S.length(); i++){
            if(jmap.contains(S.charAt(i))) count++;
        }

        return count;
    }
}

// String solution
class Solution {
    public int numJewelsInStones(String J, String S) {
        if (J == null || S==null) return 0;
        int count = 0;
        
        char stone;
        for (int i = 0; i < S.length(); i++) {
            stone = S.charAt(i);
            if(J.indexOf(stone) != -1){
                count++;
            }
        }

        return count;
    }
}

File: May-LeetCoding-Challenge/02-Jewels-And-Stones/Jewels-And-Stones.cs
// hasset solution
public class Solution {
    public int NumJewelsInStones(string J, string S) {
        HashSet<char> hashStr= new HashSet<char>();
        int counter=0;
        foreach (char c in J)
        {
            hashStr.Add(c);
        }
        foreach(char cs in S)
        {
            if (hashStr.Contains(cs))
            {
                counter++;
            }
        }
        return counter;        
    }
}
